### PR TITLE
Add support for Access behaviour, add Image.has_alpha?

### PR DESCRIPTION
**Pull request by kipcole9**
This PR adds basic support for the Access behaviour
allowing access to image bands. For example
`image[4]`.

This is helpful when crafting image pipelines
with transformations.

Note that the `pop/2` and `get_and_update/3`
callbacks are not supported and will raise an
exception if called. These cannot be readily
supported since image structures are not mutable
in Elixir space.

Also added `Image.has_alpha?/1` to return a boolean
indicating if there is an alpha band in an image.

### CODE REVIEW

**Comment by kipcole9** [comment]
Note that in addition to this PR, I am implementing basic mathematical operators (+, -, *, /, pow, cos, ...) for images for which there Access behaviour is extremely helpful.

**Comment by kipcole9** [comment]
If you're interested I think [Image](https://github.com/kipcole9/image) has made enough progress to be considered a viable experiment.  Comments and suggestions are very very welcome.

File: lib/vix/vips/image.ex [hunk 1]
```diff
@@ -42,6 +42,31 @@ defmodule Vix.Vips.Image do
   @impl Type
   def to_erl_term(ref), do: %Image{ref: ref}
 
+  # Implements the Access beahviour for Vix.Vips.Image to allow
```
Comment (akash-akya) [inline lib/vix/vips/image.ex:45]: Let's add a bit about this to `moduledoc` for the better discoverability?

**Review by akash-akya — COMMENTED** [review]
This is a nice addition. Thanks for the MR!

Just a few minor nits.

**Comment by akash-akya** [comment]
> If you're interested I think [Image](https://github.com/kipcole9/image) has made enough progress to be considered a viable experiment. Comments and suggestions are very very welcome.

@kipcole9 that looks wonderful! I'll definitely check it out more thoroughly 😄.
Happy that you are finding vix useful. Let me know if you face any issue or if you have suggestions on vix as well.

File: lib/vix/vips/image.ex (hunk 1 above)
Comment (kipcole9) [reply lib/vix/vips/image.ex]: Done, additional commit pushed.

**Pull request merged**